src/libraries/System.Private.CoreLib/src/System/Reflection/MethodBase.cs

```diff
@@ -199,7 +199,7 @@ BindingFlags invokeAttr
                             throw new ArgumentException(SR.Arg_VarMissNull, nameof(parameters));
                         }
 
-                        arg = paramInfo.DefaultValue;
+                        parameters[i] = arg = paramInfo.DefaultValue;
```

Comment:
It would be nice to do the `parameters[i]` assignment only once in all cases, ie do it in the fast path and in the slow path right before the `CheckValue` check.

BTW: The value of the fast path for default values is questionable. Reading of the default value is slow, and so we gaining very little from this fast path. You can also just delete the fast path.

Reply:
The check for "fast path" was actually required since the "slow path" did not re-check this and would have failed in Debug build due to an Assert.

The later refactor commit changed the order and now the "fast path" can occur naturally after the Type.Missing fix-ups.

Reply:
> he check for "fast path" was actually required since the "slow path" did not re-check this and would have failed in Debug build due to an Assert.

The assert in the slow path is broken. It should be removed. It is what https://github.com/dotnet/runtime/issues/73103 is about.